### PR TITLE
change data reading from form to header for download.

### LINKED ISSUE #545: Migrate request data to headers | Download method
Move all request data from query params to headers and read from there

https://0chain.slack.com/archives/G014PQ61WNT/p1645103830531899?thread_ts=1645091995.668449&cid=G014PQ61WNT

### PR DESCRIPTION
**Pull request by princeparmar**
Signed-off-by: Pradip Parmar <pradip@alfa-creator.com>

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR

Associated PRs (Link as appropriate):
- zbox: https://github.com/0chain/zboxcli/pull/177
- zwalelet:
- gosdk: https://github.com/0chain/gosdk/pull/389
- 0chain:
- system_test:

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | feature/545-migrate-querydata-to-header-download  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | feature/545-migrate-querydata-to-header-download  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | feature/545-migrate-querydata-to-header-download  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/545-migrate-querydata-to-header-download  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/545-migrate-querydata-to-header-download  |
| zboxcli  | feature/545-migrate-querydata-to-header-download  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | feature/545-migrate-querydata-to-header-download  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by cnlangzi** [comment]
@hm90121  api document should be updated for this PR. please talk to @princeparmar about detail.

**Review by cnlangzi — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/545-migrate-querydata-to-header-download  |
| zboxcli  | feature/545-migrate-querydata-to-header-download  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/blobbercore/handler/handler_test.go [hunk 1]
```diff
@@ -1610,12 +1599,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					}
 					rmData, err := json.Marshal(rm)
 					require.NoError(t, err)
-					require.NoError(t, formWriter.WriteField("read_marker", string(rmData)))
-					if err := formWriter.Close(); err != nil {
-						t.Fatal(err)
-					}
-					r, err := http.NewRequest(http.MethodPost, url.String(), body)
-					r.Header.Add("Content-Type", formWriter.FormDataContentType())
+					r, err := http.NewRequest(http.MethodPost, url.String(), nil)
```
Comment (cnlangzi) [inline code/go/0chain.net/blobbercore/handler/handler_test.go]: http.MethodPost should be updated with http.MethodGet. and please move all download tests to hanlder_download_test.go.  There are too many tests on handler_test.go. it is very difficult to read and debug.

**Review by cnlangzi — CHANGES_REQUESTED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | feature/545-migrate-querydata-to-header-download  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/545-migrate-querydata-to-header-download  |
| zboxcli  | feature/545-migrate-querydata-to-header-download  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/545-migrate-querydata-to-header-download  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by cnlangzi — APPROVED** [review]

**Pull request merged**